python/revng/tupletree.py

```diff
@@ -0,0 +1,199 @@
+from dataclasses import dataclass, fields
+from enum import Enum
+from typing import Generic, Type, TypeVar, get_args, get_origin, get_type_hints
+
+import yaml
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except ImportError:
+    from yaml import Loader, Dumper
```

Comment:
Warning?
It's not super nice but it could bring a huge performance improvement to users and they might want to know about it.

Reply:
Added a warning